jib-gradle-plugin-extension-api/build.gradle

```diff
@@ -0,0 +1,42 @@
+plugins {
+  // Is this the only option? Can't we do something simiar to Maven Extension API?
+  id 'java-gradle-plugin'
```

Comment:
I don't think you need to apply the plugin, I think you can use `gradleApi()` directly in the dependencies block? (that still makes them provided dependencies though)

Reply:
Tried `gradleApi()` in `dependencies{}`, but it doesn't even allow me to see a Gradle `Project` object in `.java` source files.

Reply:
That's weird, I feel like it should work.

https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/dsl/DependencyHandler.html#gradleApi--